addons/html_editor/static/src/components/history_dialog/history_dialog.js

```diff
@@ -101,12 +175,49 @@ export class HistoryDialog extends Component {
         this.env.services.ui.unblock();
     }
 
+    _removeExternalBlockHtml(str) {
+        const filteringRegex = /<[a-z ]+data-embedded="(?:(?!<).)+<\/[a-z]+>/gim;
+        return str.replace(
+            filteringRegex,
+            `<div class="embedded-history-dialog-placeholder">${_t("Dynamic element")}</div>`
```

Comment:
Hello, the result of this function is always wrapped in a `markup()`, which allows translation injection.

```markup`` ``` template should be used here, to properly escape the translation text.

`markup()` function should never be blindly used on the result of a function, but always inside the function instead.